plugin/storage/badger/factory.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package badger
+
+import (
+	"expvar"
+	"flag"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/dgraph-io/badger"
+	"github.com/spf13/viper"
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+
+	badgerStore "github.com/jaegertracing/jaeger/plugin/storage/badger/spanstore"
+	"github.com/jaegertracing/jaeger/storage/dependencystore"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+var (
+	// ValueLogSpaceAvailable returns the amount of space left on the value log mount point in bytes
+	ValueLogSpaceAvailable *expvar.Int
+	// KeyLogSpaceAvailable returns the amount of space left on the key log mount point in bytes
+	KeyLogSpaceAvailable *expvar.Int
+	// LastMaintenanceRun stores the timestamp of the previous maintenanceRun
+	LastMaintenanceRun *expvar.Int
+	// LastValueLogCleaned stores the timestamp of the previous ValueLogGC run
+	LastValueLogCleaned *expvar.Int
+)
+
+// Factory implements storage.Factory for Badger backend.
+type Factory struct {
+	Options *Options
+	store   *badger.DB
+	cache   *badgerStore.CacheStore
+	logger  *zap.Logger
+
+	tmpDir              string
+	maintenanceInterval time.Duration
+	maintenanceTicker   *time.Ticker
+}
+
+const (
+	defaultTickerInterval time.Duration = 5 * time.Minute
+)
+
+// NewFactory creates a new Factory.
+func NewFactory() *Factory {
+	if ValueLogSpaceAvailable == nil {
+		ValueLogSpaceAvailable = expvar.NewInt("badger_value_log_bytes_available")
```

Comment:
why not initialize these in the declaration section?

Reply:
They were originally, but I have to say I can't remember which case caused them to behave incorrectly in that scenario. Since I'm unable to replicate that issue anymore (maybe some integration test was updated in the rebases etc) I'll move them back.